### ISSUE TITLE
Card number filled by 1Password is corrupted

### ISSUE THREAD

**Issue by okovpashko** [braintree/braintree-web#262]
We have an issue with 1Password autofill: card number is filled without the last 3 digits.
Now we had to turn off autoformatting for the number field to make it work.

I've made an investigation about this issue and found that 1Password fills input directly via DOM property and then fires several events for imitating real text input.
Here is the snippet from the 1Password extension:
```
function ba(a) {
        var b = a.value
          , c = a.ownerDocument.createEvent('HTMLEvents')
          , d = a.ownerDocument.createEvent('HTMLEvents');
        a.dispatchEvent(t(a, 'keydown', 0, 0));
        a.dispatchEvent(t(a, 'keypress', 0, 0));
        a.dispatchEvent(t(a, 'keyup', 0, 0));
        d.initEvent('input', !0, !0);
        a.dispatchEvent(d);
        c.initEvent('change', !0, !0);
        a.dispatchEvent(c);
        a.blur();
        if ('' === a.value || a.dataset['com.agilebits.onepassword.initialValue'] && a.value === a.dataset['com.agilebits.onepassword.initialValue'])
                a.value = b
}
```
Even if it's not easy to understand all the details it's clearly visible what events are fired.

Next, I tried to figure out how the formatter works and I found that it unformats input after the `keypress` event if the key can mutate input value (restricted-input/lib/strategies/base.js):
```
self.inputElement.addEventListener('keypress', function (event) {
    if (keyCannotMutateValue(event)) { return; }
    self._unformatInput(event);
  });
```
But `keyCannotMutateValue` function doesn't have any logic for detecting simulated keypress.

I'm not sure if it's ok to rely on 0 in keyCode to detect 1Password autofill but I hope that this information could help you to somehow fix this issue.

**Comment by okovpashko** [comment]
A small update: 1Password fills number not without the last 3 digits but without digits exact on the place of spaces in formatted value.

**Comment by crookedneighbor** [comment]
Thanks for the update. 

I'm very :+1: on doing what we can to add support for _browser_ autofills. 

I'm a little wary about writing specific workarounds for password manager extensions, as the specific implementation could change at any time without notice. That said, your write up on what is happening looks sound to me, and I think adding support for simulated key events is 👍. 

If you'd like to PR a fix to https://github.com/braintree/restricted-input, I'm happy to review it. Otherwise, we'll work on this as we have the bandwidth.

**Comment by okovpashko** [comment]
@crookedneighbor thank you for your reply. I will write a fix and create PR.

**Comment by lilaconlee** [comment]
This is fixed in 2.31.0 of braintree-web, but we'll close this issue when this fix is in 3.x.

**Comment by crookedneighbor** [comment]
Thanks for your work on this. 3.9.0 which we just released fixes this.
